docs/writing-stories/loaders.md

```diff
@@ -0,0 +1,58 @@
+---
+title: 'Loaders'
+---
+
+Loaders are asynchronous functions that load data for a story and its [decorators](./decorators.md). A story's loaders run before the story renders, and the loaded data is passed into the story via its render context.
+
+Loaders are typically used to fetch remote API data to be used in a story, although they can be used to load any asset (e.g. lazy-loaded components).
+
+## Fetching API data
+
+Stories are isolated component examples that render internal data that's defined as part of the story or alongside the story as [args](./args.md).
+
+Loaders are useful when you need to load story data externally, e.g. from a remote API. Consider the following example that fetches a todo item for display in a todo list:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/loader-story.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+The loaded data is combined into a `loaded` field on the story context, which is the second argument to a story function. In this example we spread the story's args in first, so they take priority over the static data provided by the loader.
```

Comment:
I think we need to add some strong "DONT DO THIS UNLESS YOU HAVE A GOOD REASON" caveat or folks are going to get the wrong idea.